src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -146,6 +146,11 @@ public VisitArgumentResult(VisitResult visitResult, Optional<LocalState> stateFo
         // https://github.com/dotnet/roslyn/issues/35043: remove this when all expression are supported
         private bool _disableNullabilityAnalysis;
 
+        /// <summary>
+        /// Should be replaced by _analyzedNullabilityMapOpt if that map is always used.
+        /// </summary>
+        private PooledDictionary<BoundExpression, TypeWithState> _otherNullabilityMapOpt;
```

Comment:
>_otherNullabilityMapOpt [](start = 65, length = 23)

Consider scoping this map down to just the use case that needs it for now. If we need to expand it later, we can do so at that time.